### PR TITLE
Feat: Vehicle state errors

### PR DESCRIPTION
**Pull request by adolfoportilla**
Add `code` to vehicle state errors

### CODE REVIEW

File: lib/errors.js [hunk 1]
```diff
@@ -86,12 +86,15 @@ util.inherits(errors.ResourceNotFoundError, errors.SmartcarError);
  *
  * @extends SmartcarError
  * @param message - an error description to set
+ * @param body - A response body of Smartcar API call
+ * @param body.code - a vehicle state error code
  */
-errors.VehicleStateError = function(message) {
+errors.VehicleStateError = function(message, body) {
```
Comment (GregChan) [inline lib/errors.js]: Would it be better to have the code here since we're not using the body? No need to pass in more than we need right?

Comment (adolfoportilla) [reply lib/errors.js]: Agree, will change it!

File: lib/util.js [hunk 2]
```diff
@@ -87,7 +87,7 @@ util.catch = function(caught) {
     case 404:
       throw new errors.ResourceNotFoundError(options.uri);
     case 409:
-      throw new errors.VehicleStateError(options.uri);
+      throw new errors.VehicleStateError(options.uri, body.code);
```
Comment (gurpreetatwal) [inline lib/util.js]: this should be modified to pass `body.message` instead of the URL

File: lib/errors.js [hunk 3]
```diff
@@ -86,12 +86,14 @@ util.inherits(errors.ResourceNotFoundError, errors.SmartcarError);
  *
  * @extends SmartcarError
  * @param message - an error description to set
+ * @param code - a vehicle state error code
```
Comment (gurpreetatwal) [inline lib/errors.js]: thoughts on putting a link to the section of API reference that documents the codes for vehicle state error?

File: lib/util.js (hunk 2 above)
Comment (adolfoportilla) [reply lib/util.js]: done!

File: lib/errors.js (hunk 3 above)
Comment (adolfoportilla) [reply lib/errors.js]: I like the idea, done!

**Pull request merged**